server/plugin_release.go

```diff
@@ -750,53 +751,57 @@ func getPluginSigningSftpClient(cfg *MatterbuildConfig) (*sftp.Client, error) {
 	return sftp, nil
 }
 
-// hasExpectedPlatformBinaries verifies if plugin tar contains at least the binaries expected
-func hasExpectedPlatformBinaries(filePath string, expectedBinaries map[string]string) error {
-	f, err := os.Open(filePath)
+// findPlatformBinaries finds the binaries for which the plugin was compiled
+func findPlatformBinaries(filePath string) (map[string]string, error) {
+	tmpDir, err := os.MkdirTemp("", "platform-plugin-*")
 	if err != nil {
-		return errors.Wrap(err, "failed to setup sftp client")
+		return nil, errors.Wrap(err, "failed to create temporary directory")
 	}
-	defer f.Close()
+	defer os.RemoveAll(tmpDir) // clean up
 
-	gzf, err := gzip.NewReader(f)
+	err = unpackPlugin(filePath, tmpDir)
 	if err != nil {
-		return errors.Wrap(err, "failed to read gzip data")
+		return nil, errors.Wrap(err, "failed to unpack plugin")
 	}
 
-	foundBinaries := make(map[string]bool, len(expectedBinaries))
-	for _, v := range expectedBinaries {
-		foundBinaries[v] = false
+	dir, err := os.ReadDir(tmpDir)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read tmp dir")
 	}
 
-	tarReader := tar.NewReader(gzf)
-	for {
-		header, err := tarReader.Next()
-		if err == io.EOF {
-			break
-		}
-		if err != nil {
-			return errors.Wrap(err, "failed to read tar data")
-		}
+	// If the root of the plugin bundle consists of exactly one directory, assume the plugin
+	// is contained therein. Otherwise the root directory is expected to contain the plugin.
+	if len(dir) == 1 && dir[0].IsDir() {
+		tmpDir = filepath.Join(tmpDir, dir[0].Name())
+	}
```

Comment:
Mentioning this purely as an anecdote: this little check caused a bit of a headache to Chris a while back when we realized his local plugin bundles on MacOS included some OS specific hidden files that made this check fail mysteriously when installing the plugin.

Reply:
Indeed, we should probably port https://github.com/mattermost/mattermost-plugin-playbooks/pull/1745 back to the starter template and push it out universally.